code/sorting/bogo_sort/bogo_sorting.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+ * C Program to Implement BogoSort in an Integer Array
+ */
+#include <stdio.h>
+#include <stdlib.h>
+ 
+#define size 7
+/* Function Prototypes */
+ 
+int is_sorted(int *, int);
+void shuffle(int *, int); 
+void bogosort(int *, int);
+ 
+int main()
+{
+    int numbers[size];
+    int i;
+ 
+    printf("Enter the elements of array:");
+    for (i = 0; i < size;i++)
+    {
+        scanf("%d", &numbers[i]);
+    }
+    bogosort(numbers, size);
+    printf("The array after sorting is:");
+    for (i = 0;i < size;i++)
+    {
+        printf("%d\n", numbers[i]);
+    }
+    printf("\n");
+}
+ 
+/* Code to check if the array is sorted or not */
+int is_sorted(int *a, int n)
+{
+    while (--n >= 1)
+    {
+        if (a[n] < a[n - 1])
```

Comment:
Shouldn't this be `<=` instead of `<` so it handles duplicate values properly (or it will infinite loop)?